ql/src/java/org/apache/hadoop/hive/ql/exec/DDLPlanUtils.java

```diff
@@ -0,0 +1,962 @@
+    }
+  }
+
+  public void getAlterTableStmtDefaultConstraint(DefaultConstraint dc, Table tb, List<String> constraints) {
+    if (!DefaultConstraint.isCheckConstraintNotEmpty(dc)) {
+      return;
+    }
+    Map<String, String> colType = getTableColumnsToType(tb);
+    Map<String, List<DefaultConstraintCol>> defaultConstraints = dc.getDefaultConstraints();
+    for (String constraintName : defaultConstraints.keySet()) {
+      List<DefaultConstraintCol> defaultConstraintCols = defaultConstraints.get(constraintName);
+      for (DefaultConstraintCol col : defaultConstraintCols) {
+        ST command = new ST(ALTER_TABLE_ADD_DEFAULT_CONSTRAINT);
+        command.add(DATABASE_NAME, dc.getTableName());
+        command.add(TABLE_NAME, dc.getTableName());
+        command.add(COLUMN_NAME, col.colName);
+        command.add(COL_TYPE, colType.get(col.colName));
+        command.add(DEFAULT_VALUE, col.defaultVal);
+        constraints.add(command.render());
+      }
+    }
+  }
+
+  public void getAlterTableStmtCheckConstraint(CheckConstraint ck, List<String> constraints) {
+    if (!CheckConstraint.isCheckConstraintNotEmpty(ck)) {
+      return;
+    }
+    Map<String, List<CheckConstraint.CheckConstraintCol>> checkConstraints = ck.getCheckConstraints();
+    for (String constraintName : checkConstraints.keySet()) {
+      List<CheckConstraintCol> checkConstraintCols = checkConstraints.get(constraintName);
+      if (checkConstraintCols != null && checkConstraintCols.size() > 0) {
+        for (CheckConstraintCol col : checkConstraintCols) {
+          ST command = new ST(ALTER_TABLE_ADD_CHECK_CONSTRAINT);
+          command.add(DATABASE_NAME, ck.getDatabaseName());
+          command.add(TABLE_NAME, ck.getTableName());
+          command.add(CONSTRAINT_NAME, constraintName);
+          command.add(CHECK_EXPRESSION, col.checkExpression);
+          constraints.add(command.render());
+        }
+      }
+    }
+  }
+
+
+  public void getAlterTableStmtNotNullConstraint(NotNullConstraint nc, Table tb, List<String> constraints) {
+    if (!NotNullConstraint.isNotNullConstraintNotEmpty(nc)) {
+      return;
+    }
+    Map<String, String> colType = getTableColumnsToType(tb);
+    Map<String, String> notNullConstraints = nc.getNotNullConstraints();
+    for (String constraintName : notNullConstraints.keySet()) {
+      ST command = new ST(ALTER_TABLE_ADD_NOT_NULL_CONSTRAINT);
+      command.add(DATABASE_NAME, nc.getDatabaseName());
+      command.add(TABLE_NAME, nc.getTableName());
+      command.add(COLUMN_NAME, notNullConstraints.get(constraintName));
+      command.add(COL_TYPE, colType.get(notNullConstraints.get(constraintName)));
+      command.add(CONSTRAINT_NAME, constraintName);
+      constraints.add(command.render());
+    }
+  }
+
+  /**
+   * Add all the constraints for the given table. Will populate the constraints list.
+   *
+   * @param tb
+   */
+  public List<String> populateConstraints(Table tb, Set<String> all_tbl) {
```

Comment:
Rename all_tbl to allTableNames? Same in getAlterTableStmtForeignKeyConstraint() as well.